settings.php

```diff
@@ -0,0 +1,423 @@
+				</div>
+			</div>
+			<div class="box-footer">
+				<input type="hidden" name="field" value="DNS">
+				<button type="submit" class="btn btn-primary pull-right">Save</button>
+			</div>
+			</form>
+		</div>
+	</div>
+	<div class="col-md-6">
+<?php
+	// Query logging
+	if(isset($setupVars["QUERY_LOGGING"]))
+	{
+		if($setupVars["QUERY_LOGGING"] == 1)
+		{
+			$piHoleLogging = true;
+		}
+		else
+		{
+			$piHoleLogging = false;
+		}
+	} else {
+		$piHoleLogging = true;
+	}
+?>
+		<div class="box box-primary">
+			<div class="box-header with-border">
+				<h3 class="box-title">Query Logging (size of log <?php echo formatSizeUnits(filesize("/var/log/pihole.log")); ?>)</h3>
+			</div>
+			<div class="box-body">
+				<form role="form" method="post">
+				<p>Current status:
+				<?php if($piHoleLogging) { ?>
+					Enabled (recommended)
+				<?php }else{ ?>
+					Disabled
+				<?php } ?></p>
+				<input type="hidden" name="field" value="Logging">
+				<?php if($piHoleLogging) { ?>
+					<p>Note that disabling will render graphs on the web user interface useless</p>
+				<?php } ?>
+				</form>
+			</div>
+			<div class="box-footer">
+				<input type="hidden" name="field" value="DNS">
+				<?php if($piHoleLogging) { ?>
+					<input type="hidden" name="action" value="Disable">
+					<button type="submit" class="btn btn-primary pull-right">Disable query logging</button>
+				<?php } else { ?>
+					<input type="hidden" name="action" value="Enable">
+					<button type="submit" class="btn btn-primary pull-right">Enable query logging</button>
+				<?php } ?>
+			</div>
+		</div>
+<?php
+	// Exluded domains
+	if(isset($setupVars["API_EXCLUDE_DOMAINS"]))
+	{
+		$excludedDomains = explode(",", $setupVars["API_EXCLUDE_DOMAINS"]);
+	} else {
+		$excludedDomains = "";
+	}
+
+	// Exluded clients
+	if(isset($setupVars["API_EXCLUDE_CLIENTS"]))
+	{
+		$excludedClients = explode(",", $setupVars["API_EXCLUDE_CLIENTS"]);
+	} else {
+		$excludedClients = "";
+	}
+
+	// Exluded clients
+	if(isset($setupVars["API_QUERY_LOG_SHOW"]))
+	{
+		$queryLog = $setupVars["API_QUERY_LOG_SHOW"];
+	} else {
+		$queryLog = "all";
+	}
+?>
+		<div class="box box-success">
+			<div class="box-header with-border">
+				<h3 class="box-title">API</h3>
+			</div>
+			<form role="form" method="post">
+			<div class="box-body">
+				<h4>Top Lists</h4>
+				<p>Exclude the following domains from being shown in</p>
+				<div class="col-lg-6">
+					<div class="form-group">
+					<label>Top Domains / Top Advertisers</label>
+					<textarea name="domains" class="form-control" rows="4" placeholder="Enter one domain per line"><?php foreach ($excludedDomains as $domain) { echo $domain."\n"; } ?></textarea>
+					</div>
+				</div>
+				<div class="col-lg-6">
+					<div class="form-group">
+					<label>Top Clients</label>
+					<textarea name="clients" class="form-control" rows="4" placeholder="Enter one domain per line"><?php foreach ($excludedClients as $client) { echo $client."\n"; } ?></textarea>
```

Comment:
Will clients be IP addresses? If so, change "domain" to "IP address" in the placeholder.